interface/usergroup/user_admin.php

```diff
@@ -129,6 +129,18 @@ function submitform() {
         }
 
     }//If pwd null ends here
+
+    // Valiate Google email (if provided)
+    if(document.forms[0].google_signin_email.value != ""){
+        // https://owasp.org/www-community/OWASP_Validation_Regex_Repository
+        var mailformat = /^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}$/;
+        if(!document.forms[0].google_signin_email.value.match(mailformat)) {
+            flag=1;
+            alert(<?php echo xlj('Google email provided is invalid/not properly formatted (e.g. first.last@gmail.com)') ?>);
+            return false;
+        }
+    }
```

Comment:
Instead of duplicating code, rec placing a js function for checking email addresses in the library/js/utility.js script. This script is automatically brought into these php scripts, so you will be able to call the js function from them.

Also, are email addressing strictly ascii characters or can they include accented characters and chinese characters etc? (I don't know the answer to this)

Reply:
btw, the nice thing about centralizing this kind of thing is ability to adjust the function as needed in the future.

Reply:
Great tip! Thanks I have refactored the code and pushed an updated commit.